### PR TITLE
refactor(BUX-402): code refactorization before introducing PIKE demo

### PR DESCRIPTION
**Pull request by chris-4chain**
Changes:
- AxiosProvider sets `apiUrl` and interceptors before app is rendered
- The interceptor for common options like: withCredentials or CacheControl
- `apiUrl` hook is not necessary anymore (which simplified data fetching in useEffects)
- reduced number of files in `/api/requests` - each file for each url-group;
- PaginationParams type
- using `params` axios option to automatically generate "query string" `(e.g.?cond=value)` in url

# Pull Request Checklist

- [x] 📖 I created my PR using provided: [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-web-frontend/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-web-frontend/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

**Review by wregulski — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -61,7 +60,6 @@ const ROUTES_AUTHENTICATED = [
 
 export const App = () => {
   const { authorization, setAuthorization } = useAuthorization()
-  const apiUrl = useApiUrl()
   const navigate = useNavigate()
   const [loading, setLoading] = useState(true)
```
Comment (Nazarii-4chain) [inline src/App.tsx:64]: There is a great lib names React Query. It helps handle state, errors and loadings

File: src/App.tsx [hunk 2]
```diff
@@ -70,41 +68,39 @@ export const App = () => {
...
     setError('')
-    apiUrl &&
-      getUser(apiUrl)
-        .then((response) => {
-          const currentUserData: LoggedInUser = {
-            email: response.email,
-            paymail: response.paymail,
-            balance: response.balance,
-          }
+    getUser()
```
Comment (Nazarii-4chain) [inline src/App.tsx:71]: Better to use async/await approach with try/catch. It is more readable

File: src/components/AccountSummary/AccountSummary.tsx [hunk 3]
```diff
@@ -24,40 +23,38 @@ interface AccountDetails {
...
-            balance: response.balance,
-            email: response.email,
-            paymail: response.paymail,
-          }
-          setDetails(accountDetails)
-        })
-        .catch((error) => {
-          let errorMsg
+    getUser()
+      .then((response) => {
```
Comment (Nazarii-4chain) [inline src/components/AccountSummary/AccountSummary.tsx:34]: Same here. Async/await approach

File: src/components/Modal/_modals/TransactionConfirmModal/TransactionConfirmModal.tsx [hunk 4]
```diff
@@ -62,45 +60,42 @@ export const TransactionConfirmModal: FC<TransactionConfirmModalProps> = ({
...
-              return
-            }
-
-            setErrors(
-              error.response.data
-                ? error.response.data
-                : 'Transfer was not sent. Please verify transfer data and try once again. If problem will happen again, contact with our support.'
-            )
+    sendTransaction(newTransactionData)
+      .then(() => {
```
Comment (Nazarii-4chain) [inline src/components/Modal/_modals/TransactionConfirmModal/TransactionConfirmModal.tsx:64]: Same here. Async/await approach

File: src/components/Modal/_modals/TransactionDetailsModal/TransactionDetailsModal.tsx [hunk 5]
```diff
@@ -26,23 +25,20 @@ export const TransactionDetailsModal: FC<TransactionDetailsProps> = ({ open, pri
...
-        })
-        .catch((error) => {
-          const errorMsg = error.response.data ? error.response.data : 'Something went wrong... Please try again later'
-          errorMsg && setErrors(errorMsg)
-        })
-        .finally(() => {
-          setLoading(false)
-        })
-  }, [apiUrl, id])
+    getTransactionsDetails(id)
```
Comment (Nazarii-4chain) [inline src/components/Modal/_modals/TransactionDetailsModal/TransactionDetailsModal.tsx:30]: Same here. Async/await approach

File: src/App.tsx (hunk 1 above)
Comment (chris-4chain) [reply src/App.tsx:64]: Will be considered in the future. For now it's out of scope of this PR.

**Pull request merged**